torch/onnx/symbolic_opset9.py

```diff
@@ -1323,6 +1323,9 @@ def layer_norm(g, input, normalized_shape, weight, bias, eps, cudnn_enable):
     eps_cst = g.op("Constant", value_t=torch.tensor(eps))
 
     mean = g.op("ReduceMean", input, axes_i=axes)
+    if mean.type().scalarType() == 'Half':
+        two_cst = g.op("Constant", value_t=torch.tensor(2))
```

Comment:
can you try create these two in a way for them to be recognized in scalar_type_analysis pass as a scalar?
And then even better create a helper function for scalar creation.

Reply:
if they are recognized as scalar in that pass, their dtype will have lower priority, and will be overwritten with say half in this case. This way it's more consistent with pt casting logic.